### PR TITLE
Add auto-update functionality to all data sources

### PR DESCRIPTION
**Pull request by c-w**
This pull request adds an optional property to the dashboard data sources: autoUpdateIntervalMs. If this property is set, the data source will refetch its query at the interval defined by the property. For example, to re-fetch the results for an application insights query twice per second, we'd start our data source configuration like this:

image

Paired with @eladiw

### CODE REVIEW

File: src/data-sources/plugins/DataSourcePlugin.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export interface IDataSourcePlugin {
   };
 
   bind (actionClass: any): void;
-  updateDependencies (dependencies: IDictionary, args: IDictionary, callback: () => void): void;
+  _updateDependencies (dependencies: IDictionary, args: IDictionary, callback: () => void): void;
```
Comment (itye-msft) [inline src/data-sources/plugins/DataSourcePlugin.ts]: I believe that our naming convention for class functions don't use underscore prefix "_"

Comment (c-w) [inline src/data-sources/plugins/DataSourcePlugin.ts]: This was done to indicate a private wrapper. I've renamed to make this more explicit in c4c5c0e.

**Comment by morsh** [comment]
@itye-msft - since these are abstract methods and are implemented as part of various components, updateDependenciesInternal does not work.

Please rename to something more descriptive like "dependenciesUpdated"

**Comment by itye-msft** [comment]
agreed and done.

**Pull request merged**